### PR TITLE
Bug/en 2812 round index int64

### PR DESCRIPTION
**Pull request by camilbancioiu**
Change the type of the Round Index throughout the project from `int32` to `int64`. Integer overflows occured when stress-testing the consensus with rounds less than 1 second. Also, until we implement end-of-Epoch events to reset the Round Index in each Epoch, there is a small risk of running out of values for the Round Index - because it was calculated from the Unix Epoch, rounds of 4 seconds would need indexes larger than MAXINT on 32 bits in some decades from now. As networking becomes faster and round durations potentially shorter in the near future, this would have become a serious issue. 

While end-of-Epoch events would solve this issue by default, they are not yet implemented. Thus we should switch to `int64` for the Round Index.

### CODE REVIEW

File: data/block/metaBlock_test.go [hunk 1]
```diff
@@ -2,10 +2,11 @@ package block_test
 
 import (
 	"bytes"
-	"github.com/ElrondNetwork/elrond-go/sharding"
 	"math/big"
 	"testing"
 
+	"github.com/ElrondNetwork/elrond-go/sharding"
+
```
Comment (sasurobert) [inline data/block/metaBlock_test.go]: arrange imports

File: integrationTests/multiShard/block/executingVMCalls_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package block
```
Comment (sasurobert) [inline integrationTests/multiShard/block/executingVMCalls_test.go]: delete this file

File: process/mock/txProcessorMock.go [hunk 3]
```diff
@@ -1,19 +1,20 @@
 package mock
 
 import (
-	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
 	"math/big"
 
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+
```
Comment (sasurobert) [inline process/mock/txProcessorMock.go]: delete empty line

File: process/smartContract/export_test.go [hunk 4]
```diff
@@ -1,12 +1,13 @@
 package smartContract
 
 import (
-	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
 	"math/big"
 
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
```
Comment (sasurobert) [inline process/smartContract/export_test.go]: goimports on this file

File: process/smartContract/process.go [hunk 5]
```diff
@@ -15,7 +15,7 @@ import (
 	"github.com/ElrondNetwork/elrond-go/marshal"
 	"github.com/ElrondNetwork/elrond-go/process"
 	"github.com/ElrondNetwork/elrond-go/sharding"
-	"github.com/ElrondNetwork/elrond-vm-common"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
```
Comment (sasurobert) [inline process/smartContract/process.go]: vmcommon alias not needed. verify all added aliases.

File: process/smartContract/process_test.go [hunk 6]
```diff
@@ -3,15 +3,16 @@ package smartContract
 import (
 	"bytes"
 	"crypto/rand"
-	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
 	"math/big"
 	"testing"
 
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+
```
Comment (sasurobert) [inline process/smartContract/process_test.go]: goimports

File: process/transaction/process.go [hunk 7]
```diff
@@ -2,9 +2,10 @@ package transaction
 
 import (
 	"bytes"
-	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
 	"math/big"
 
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+
```
Comment (sasurobert) [inline process/transaction/process.go]: goimports

File: process/transaction/process_test.go [hunk 8]
```diff
@@ -4,11 +4,12 @@ import (
...
 	"crypto/rand"
 	"errors"
-	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
-	"github.com/ElrondNetwork/elrond-go/process/smartContract"
 	"math/big"
 	"testing"
 
+	"github.com/ElrondNetwork/elrond-go/data/smartContractResult"
+	"github.com/ElrondNetwork/elrond-go/process/smartContract"
+
```
Comment (sasurobert) [inline process/transaction/process_test.go]: goimports

File: data/block/metaBlock_test.go (hunk 1 above)
Comment (camilbancioiu) [reply data/block/metaBlock_test.go]: Fixed.

File: integrationTests/multiShard/block/executingVMCalls_test.go (hunk 2 above)
Comment (camilbancioiu) [reply integrationTests/multiShard/block/executingVMCalls_test.go]: Deleted.

File: process/mock/txProcessorMock.go (hunk 3 above)
Comment (camilbancioiu) [reply process/mock/txProcessorMock.go]: Fixed.

File: process/smartContract/export_test.go (hunk 4 above)
Comment (camilbancioiu) [reply process/smartContract/export_test.go]: Fixed.

File: process/smartContract/process.go (hunk 5 above)
Comment (camilbancioiu) [reply process/smartContract/process.go]: Fixed.

File: process/smartContract/process_test.go (hunk 6 above)
Comment (camilbancioiu) [reply process/smartContract/process_test.go]: Fixed.

File: process/transaction/process.go (hunk 7 above)
Comment (camilbancioiu) [reply process/transaction/process.go]: Fixed.

File: process/transaction/process_test.go (hunk 8 above)
Comment (camilbancioiu) [reply process/transaction/process_test.go]: Fixed.

File: integrationTests/mock/blockProcessorMock.go [hunk 9]
```diff
@@ -1,9 +1,10 @@
 package mock
 
 import (
+	"time"
+
 	"github.com/ElrondNetwork/elrond-go/data/block"
 	"github.com/ElrondNetwork/elrond-go/marshal"
-	"time"
```
Comment (raduchis) [inline integrationTests/mock/blockProcessorMock.go]: Could remove the empty line

**Pull request merged**